pkg/cli/admin/release/extract_tools.go

```diff
@@ -1286,3 +1287,32 @@ func newIncluder(config manifestInclusionConfiguration) includer {
 		return m.Include(config.ExcludeIdentifier, config.RequiredFeatureSet, config.Profile, config.Capabilities, config.Overrides)
 	}
 }
+
+// ManifestReceiver has a TarEntryCallback function which can be used to as a callback to ExtractOptions.TarEntryCallback.
+// It feeds the downstream manifestsCallback
+// * with the manifests from every file whose name is not in skipNames, OR
+// * with the reader that contains the content of each file whose name is in skipNames.
+// All the errors encountered when parsing the manifests are collected in ManifestErrs.
+type ManifestReceiver struct {
+	manifestsCallback func(filename string, manifests []manifest.Manifest, reader io.Reader) (cont bool, err error)
+	skipNames         sets.Set[string]
```

Comment:
The `skipNames` naming is a bit confusing to me; "skip" makes me think they are skipped entirely, but they are actually just not parsed as manifests. And actually they are anti-skipped because they would otherwise be skipped because they are not json/yaml files.